erigon-lib/state/appendable_extras/snap_config.go

```diff
@@ -0,0 +1,98 @@
+package entity_extras
+
+import (
+	"fmt"
+
+	"github.com/erigontech/erigon-lib/chain/snapcfg"
+	"github.com/erigontech/erigon-lib/downloader/snaptype"
+)
+
+// aggregate set level snapshot creation config
+// entities in the set should have same config
+type SnapshotCreationConfig struct {
+	// number of RootNums per step
+	// should be same for all entity in an entity set
+	RootNumPerStep uint64
+
+	// how many (root) entities to leave in db (and not consider for freezing) this is needed
+	// since blockchains reorg and so we don't freeze latest entities.
+	SafetyMargin uint64
+
+	// progressively merge smaller files into large ones.
+	// maximum size (merge limit) is the last element of MergeStages
+	// decreasing order expected, each step is a multiple of the previous one
+	// e.g. [1000, 20000, 600000] --> first stage creates files of size 1000; then 20 of these merged to
+	// create size 10000; then 30 of these merged to create size 100000
+	// each must be divisible by `RootNumPerStep`
+	// last stage is considered to be "frozen".
+	MergeStages []uint64
```

Comment:
it doesn't capture the possibly infinite merging of accounts domain. But that's okay - we just need to have sufficient number of entries here to cover us for next 100 years or something, and it won't make a difference.

File is considered "frozen" when it is at the last MergeStage.